[Serve] [Docs] Add Serve REST API Schema to Serve API Docs

shrekris-anyscale opened the pull request:
<!-- Thank you for your contribution! Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this solves. -->

#25691 updates the Serve CLI and REST API. This change adds the new REST API schema to the documentation, including info about endpoints, REST requests, and the input and output JSON schemas.

Link to this change's documentation: https://ray--25786.org.readthedocs.build/en/25786/serve/package-ref.html#serve-rest-api

## Related issue number

<!-- For example: "Closes #1234" -->

N/A

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - N/A: this is a documentation change.

simon-mo commented:
re: httpdomain
<img width="768" alt="image" src="https://user-images.githubusercontent.com/21118851/174160889-ef22b705-fdb4-43be-9411-345daf53717d.png">

if we are _just_ rendering these with the new package, it seems a little bit excessive? I was under the impression there are more ux features as well. But looks like just simple headings? 


re: CLI, great job!

edoakes commented on doc/source/serve/package-ref.md:
```diff
@@ -45,3 +45,63 @@
+.. http:GET:: "/api/serve/deployments/"
+    :noindex:
+
+        Gets latest config that Serve has received. This config represents the current goal state for the Serve application. Starts a Serve application on the Ray cluster if it's not already running. See Config Schema for the output schema.
```
link to "config schema" and don't capitalize. also goes for other instances

shrekris-anyscale replied:
Good catch– I made the updates.

edoakes commented on doc/source/serve/package-ref.md:
```diff
@@ -45,3 +45,63 @@
+.. http:PUT:: "/api/serve/deployments/"
+    :noindex:
+
+        Uploads config for Serve application to deploy. Starts a Serve application on the Ray cluster if it's not already running. See Config Schema for the request schema.
```
```suggestion
        Declaratively deploys the Serve application. Starts Serve on the Ray cluster if it's not already running. See Config Schema for the request schema.
```

edoakes reviewed (commented):
just nits

shrekris-anyscale commented:
> if we are just rendering these with the new package, it seems a little bit excessive? I was under the impression there are more ux features as well. But looks like just simple headings?

@simon-mo Good points– I removed the dependency and replaced the formatting with Markdown. I've also added some example requests and responses to all the REST API methods.

edoakes commented:
@shrekris-anyscale lint failure
